plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageSettings.java

```diff
@@ -51,6 +51,12 @@
         key -> SecureSetting.secureString(key, null));
 
     /**
+     * Azure endpoint suffix. Default to core.windows.net (CloudStorageAccount.DEFAULT_DNS).
+     */
+    public static final AffixSetting<SecureString> ENDPOINT_SUFFIX_SETTING = Setting.affixKeySetting(PREFIX, "endpoint_suffix",
```

Comment:
Why does this need to be a secure setting?

Reply:
Thanks! I think a normal string is OK. Shall I merge with master for review?